packages/notices/src/store/utils/on-sub-key.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * Higher-order reducer creator which creates a combined reducer object, keyed
+ * by a property on the action object.
+ *
+ * @param {string} actionProperty Action property by which to key object.
+ *
+ * @return {Function} Higher-order reducer.
+ */
+export const onSubKey = ( actionProperty ) => ( reducer ) => ( state = {}, action ) => {
```

Comment:
My second thought is that it might be also a good idea to move it to `data` package next to other helpers like `combineReducers` as it has a very similar application. Eventually, we could extract those utilities to their own package. It's very unlikely that we will use it without reducers :)

Reply:
I think I like that, especially if even the main fear would be bloat for a size-conscious module developer, it'd be easily omitted by tree shaking.

Reply:
> I think I like that, especially if even the main fear would be bloat for a size-conscious module developer, it'd be easily omitted by tree shaking.

Let's keep this to a follow-up pull request.

Reply:
Agreed, this one is big enough 👍